met-web/src/components/engagement/form/EngagementWidgets/WhoIsListeningForm.tsx

```diff
@@ -8,15 +8,34 @@ import { postWidgetItems } from 'services/widgetService';
 import { WidgetDrawerContext } from './WidgetDrawerContext';
 import ContantInfoPaper from './ContactInfoPaper';
 import { WidgetType } from 'models/widget';
+import update from 'immutability-helper';
 
 const WhoIsListeningForm = () => {
-    const { handleWidgetDrawerOpen, handleAddContactDrawerOpen, loadingContacts, contacts, widgets } =
+    const { handleWidgetDrawerOpen, handleAddContactDrawerOpen, loadingContacts, contacts, widgets, clearSelected } =
         useContext(WidgetDrawerContext);
     const dispatch = useAppDispatch();
     const [selectedContact, setSelectedContact] = useState<Contact | null>(null);
     const [addedContacts, setAddedContacts] = useState<Contact[]>([]);
     const [addingWidgetItems, setAddingWidgetItems] = useState(false);
 
+    useEffect(() => {
+        const filteredContacts = contacts.filter((c: Contact) =>
+            addedContacts.map((contact) => contact.id).includes(c.id),
+        );
+        setAddedContacts(filteredContacts);
+    }, [contacts]);
+
+    const moveContact = useCallback((dragIndex: number, hoverIndex: number) => {
```

Comment:
Question here, how are we saving the sorting order after the dragging happens? So like if contacts where A then B then C, user dragged B above A. When we open drawer again will we see B, A C?

Reply:
sorting order is important ; each contact has to be shown in the order they were dragged... I am okay to handle it in next PR